packages/textmate-grammars/src/browser/phphtml.ts

```diff
@@ -0,0 +1,93 @@
+ ********************************************************************************/
+
+import { LanguageGrammarDefinitionContribution, TextmateRegistry } from '@theia/monaco/lib/browser/textmate';
+import { injectable } from 'inversify';
+
+@injectable()
+export class PhpHtmlGrammarContribution implements LanguageGrammarDefinitionContribution {
+    readonly id = 'php';
+    readonly scopeName = 'text.html.php';
+
+    readonly config: monaco.languages.LanguageConfiguration = {
+        comments: {
+            lineComment: '//', // '#'
+            blockComment: ['/*', '*/']
+        },
+        brackets: [
+            ['{', '}'],
+            ['[', ']'],
+            ['(', ')']
+        ],
+        autoClosingPairs: [
+            { open: '{', close: '}', notIn: ['string'] },
+            { open: '[', close: ']', notIn: ['string'] },
+            { open: '(', close: ')', notIn: ['string'] },
+            { open: '\'', close: '\'', notIn: ['string', 'comment'] },
+            { open: '"', close: '"', notIn: ['string', 'comment'] },
+            { open: '/**', close: ' */', notIn: ['string'] }
+        ],
+        surroundingPairs: [
+            { open: '{', close: '}' },
+            { open: '[', close: ']' },
+            { open: '(', close: ')' },
+            { open: '\'', close: '\'' },
+            { open: '"', close: '"' },
+            { open: '`', close: '`' }
+
+        ],
+        indentationRules: {
+            increaseIndentPattern: new RegExp('({(?!.+}).*|\\(|\\[|((else(\\s)?)?if|else|for(each)?|while|switch).*:)\\s*(/[/*].*)?$'),
+            decreaseIndentPattern: new RegExp('^(.*\\*\\/)?\\s*((\\})|(\\)+[;,])|(\\][;,])|\\b(else:)|\\b((end(if|for(each)?|while|switch));))')
+        },
+        folding: {
+            markers: {
+                start: /\\s*(#|\/\/)region\\b/,
+                end: /^\\s*(#|\/\/)endregion\\b/
+            }
+        }
+    };
+
+    registerTextmateLanguage(registry: TextmateRegistry) {
+        monaco.languages.register({
+            id: this.id,
+            'extensions': [
+                '.html', '.htm', '.shtml', '.xhtml', '.mdoc', '.jsp', '.asp', '.aspx', '.jshtm'
+            ],
+            'aliases': [
+                'HTML', 'htm', 'html', 'xhtml'
+            ],
+            'firstLine': '^#!\\s*/.*\\bphp\\b',
+            'mimetypes': [
+                'text/html', 'text/x-jshtm', 'text/template', 'text/ng-template'
+            ]
+        });
+
+        monaco.languages.setLanguageConfiguration(this.id, this.config);
+
+        const phpHtmlGrammar = require('../../data/php.html.tmLanguage.json');
+        registry.registerTextmateGrammarScope(this.scopeName, {
+            async getGrammarDefinition() {
+                return {
+                    format: 'json',
+                    content: phpHtmlGrammar
+                };
+            }
+        });
+
+        registry.mapLanguageIdToTextmateGrammar(this.id, this.scopeName);
```

Comment:
I don't thin we support multiple scopes per a language. It looks like we need to drop phphtml support till it is implemented. Could you file an issue to support multiple scopes?

Reply:
verified, it breaks:
<img width="1737" alt="Screen Shot 2019-04-02 at 06 24 17" src="https://user-images.githubusercontent.com/3082655/55376275-04360e80-5510-11e9-9579-0a155bc2a917.png">

php grammar gets overridden by phphtml, we need to pick one or another scope name for now

Reply:
OK. I'll remove phphtml for now and open an issue to support multiple scopes